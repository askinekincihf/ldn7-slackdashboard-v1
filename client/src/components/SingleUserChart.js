import { Line, Bar } from "react-chartjs-2";

const SingleUserChart = (props) => {
	const state = {
		data: {
			labels: props.label,
			datasets: [
				{
					label: "Messages",
					backgroundColor: "green",
					borderColor: "green",
					borderWidth: 1,
					hoverBackgroundColor: "lightGreen",
					hoverBorderColor: "lightGreen",
					data: props.messagesDataSet,
				},

				{
					label: "Reactions",
					backgroundColor: "orange",
					borderColor: "orange",
					borderWidth: 1,
					hoverBackgroundColor: "rgba(255,99,132,0.4)",
					hoverBorderColor: "rgba(255,99,132,1)",
					data: props.reactionsDataSet,
				},
			],
		},
	};
	const options = {
		responsive: true,
		legend: {
			display: false,
		},
		type: "bar",
	};
	return (
		<div className="chart">
			<Line data={state.data} options={options} />
		</div>
	);
};

export default SingleUserChart;
