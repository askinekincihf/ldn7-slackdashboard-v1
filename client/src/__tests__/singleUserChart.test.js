import React from "react";
import renderer from "react-test-renderer";
import SingleUserChart from "../components/SingleUserChart";

describe("SingleUserChart", () => {
	it("should render a chart for a single user correctly", () => {
		const component = renderer.create(<SingleUserChart />).toJSON();

		expect(component).toMatchSnapshot();
	});
});
