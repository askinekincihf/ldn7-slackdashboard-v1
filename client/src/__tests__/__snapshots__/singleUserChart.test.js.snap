// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleUserChart should render a chart for a single user correctly 1`] = `
<div
  className="singleUserChart"
>
  <canvas
    data-testid="canvas"
    height={150}
    onClick={[Function]}
    role="img"
    width={300}
  />
</div>
`;
